catalog/catalog.go

```diff
@@ -146,9 +147,9 @@ type Catalog interface {
 	// identifiers containing the information required to load the table via that catalog.
 	ListTables(ctx context.Context, namespace table.Identifier) ([]table.Identifier, error)
 	// LoadTable loads a table from the catalog and returns a Table with the metadata.
-	LoadTable(ctx context.Context, identifier table.Identifier, props iceberg.Properties) (*table.Table, error)
+	LoadTable(ctx context.Context, identifier table.Identifier) (*table.Table, error)
```

Comment:
Why remove properties parameter that allowed for overriding props during load?

Reply:
I have restored the props field, I simply noticed that it was always set to nil. But I agree that might be useful.